sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/NotificationHookLiveTests.cs

```diff
@@ -0,0 +1,389 @@
+                hookToCreate.Headers = headers;
+            }
+
+            await using var disposableHook = await DisposableNotificationHook.CreateHookAsync(adminClient, hookToCreate);
+
+            NotificationHook createdHook = await adminClient.GetHookAsync(disposableHook.Id);
+
+            Assert.That(createdHook.Id, Is.EqualTo(disposableHook.Id));
+            Assert.That(createdHook.Name, Is.EqualTo(hookName));
+            Assert.That(createdHook.Administrators, Is.Not.Null);
+            Assert.That(createdHook.Administrators.Single(), Is.Not.Null.And.Not.Empty);
+
+            var createdWebHook = createdHook as WebNotificationHook;
+
+            Assert.That(createdWebHook, Is.Not.Null);
+            Assert.That(createdWebHook.Endpoint, Is.EqualTo(endpoint));
+
+            if (populateOptionalMembers)
+            {
+                Assert.That(createdHook.Description, Is.EqualTo(description));
+                Assert.That(createdHook.ExternalLink, Is.EqualTo("http://fake.endpoint.com"));
+                // TODO: add certificate key validation (https://github.com/Azure/azure-sdk-for-net/issues/17485).
+                Assert.That(createdWebHook.CertificatePassword, Is.EqualTo("certPassword"));
+                Assert.That(createdWebHook.Username, Is.EqualTo("fakeUsername"));
+                Assert.That(createdWebHook.Password, Is.EqualTo("fakePassword"));
+                Assert.That(createdWebHook.Headers, Is.EquivalentTo(headers));
+            }
+            else
+            {
+                Assert.That(createdHook.Description, Is.Empty);
+                Assert.That(createdHook.ExternalLink, Is.Empty);
+                Assert.That(createdWebHook.CertificateKey, Is.Empty);
+                Assert.That(createdWebHook.CertificatePassword, Is.Empty);
+                Assert.That(createdWebHook.Username, Is.Empty);
+                Assert.That(createdWebHook.Password, Is.Empty);
+                Assert.That(createdWebHook.Headers, Is.Not.Null.And.Empty);
+            }
+        }
+
+        /// <param name="updateWithNewInstance">
+        /// Users have two ways of creating a <see cref="NotificationHook"/>, and this parameter is used to test the Update
+        /// method with both. If <c>true</c>, the instance passed to Update is created from scratch. If <c>false</c>, the
+        /// instance passed to Update is obtained from the Get method.
+        /// </param>
+        /// <param name="updateAllMembers">
+        /// When <c>true</c>, updates all possible members to make sure values are being passed and returned correctly.
+        /// When <c>false</c>, only updates one member, and makes sure that other values were not altered.
+        /// </param>
+        [RecordedTest]
+        [TestCase(true, true)]
+        [TestCase(true, false)]
+        [TestCase(false, true)]
+        [TestCase(false, false)]
+        public async Task UpdateEmailNotificationHook(bool updateWithNewInstance, bool updateAllMembers)
```

Comment:
I agree that having two boolean variables makes it hard to track. Do you think it would be okay to keep the `updateWithNewInstance` parameter, but having two tests (`Update...HookWithMinimumSetup` and `Update...HookWithAllMembersSet`)?

Reply:
hmm personally I prefer 4 separate ones. but if you want to aggregate, maybe with optional parameters (like other tests) and having the `UpdateWithNewInstance` as separate methods? thinking that this one involves an extra service call, so if it fails, easier to figure out why by having that call isolated.

Reply:
Separated this and other tests with bool parameters into different tests. Usually:
`<MethodName>WithMinimumSetup` and `<MethodName>WithOptionalMembers`. If there's only one or a few related optional members, the chosen name is more specific (such as `...WithOptionalDimensionFilter`).

The Update tests were separated into `...WithMinimumSetupAndGetInstance`, `...WithMinimumSetupAndNewInstance`, `...WithEveryMemberAndGetInstance`, and `...WithEveryMemberAndNewInstance`.

Apologies for the huge amount of changes. I'm basically splitting the logic, so there should be no concerns in most of the files.